### PR TITLE
Batch block data requests from fuel-core GraphQL API

### PR DESCRIPTION
**Pull request by Dentosal**
Fetch blocks at groups of 64, alongside with receipts. Reduces overhead quite a bit.

~~Blocked by https://github.com/FuelLabs/fuel-core/pull/1732~~

### CODE REVIEW

**Review by Salka1988 — DISMISSED** [review]
This is great!

File: firehose-extract/src/types.rs [hunk 1]
```diff
@@ -281,6 +330,123 @@ impl From<&FuelStorageSlot> for StorageSlot {
...
+        //             sub_id: (*receipt.sub_id.as_ref().unwrap().0 .0).into(),
+        //             contract_id: (*receipt.contract_id.as_ref().unwrap().0 .0).into(),
+        //             val: receipt.val.unwrap().into(),
+        //             pc: receipt.pc.unwrap().into(),
+        //             is: receipt.is.unwrap().into(),
+        //         }),
+        //     }),
+        // }
+    }
+}
```
[the remark this answers is not in the document]
Comment (Dentosal) [reply firehose-extract/src/types.rs:338]: Good catch. That was used to work around fuel-core-client v0.22.2, and is not useful anymore. Removed in 877522d.

File: firehose-extract/build.rs [hunk 2]
```diff
@@ -1,8 +1,14 @@
+use std::fs;
 use std::io::Result;
+
 fn main() -> Result<()> {
     prost_build::compile_protos(
         &["type.proto"],
         &["../firehose-fuel/proto/sf/fuel/type/v1/"],
     )?;
+
+    fs::write("target/schema.sdl", fuel_core_client::SCHEMA_SDL)
```
Comment (xgreenx) [inline firehose-extract/build.rs:10]: Hmm, you can publish `fuel-core-client-ext` and use it directly instead of duplicating the code=)

Comment (Dentosal) [reply firehose-extract/build.rs:10]: https://github.com/FuelLabs/fuel-core-client-ext/pull/1

Comment (Dentosal) [reply firehose-extract/build.rs:10]: https://github.com/FuelLabs/fuel-core-client-ext/pull/2

Comment (Dentosal) [reply firehose-extract/build.rs]: Ok done

**Review by Salka1988 — APPROVED** [review]

**Pull request merged**